src/core/currency/wallet/currency-wallet-reducer.js

```diff
@@ -278,8 +278,9 @@ const currencyWalletInner: FatReducer<
     state: EdgeStakingStatus = { stakedAmounts: [] },
     action: RootAction
   ): EdgeStakingStatus {
+    // Copy the object to guard against engines that mutate their data:
     return action.type === 'CURRENCY_ENGINE_CHANGED_STAKING'
-      ? action.payload.stakingStatus
+      ? { ...action.payload.stakingStatus }
```

Comment:
We ended up patching this in the plugin, but we should include this in the core to future proof this behavior. 👍

Reply:
Actually, this will only work for the top-level object. If you mutate the array inside, then we would have to splat that too...

Maybe this could be `JSON.parse(JSON.stringify())`, which would kill the problem dead. Except, it won't fix date objects, so... maybe it's better if we just let the plugins do the right thing.